packages/aila/src/core/document/AilaDocument.ts

```diff
@@ -11,73 +11,70 @@ import {
 import type { LooseLessonPlan } from "../../protocol/schema";
 import type { AilaDocumentService, AilaServices } from "../AilaServices";
 import type { Message } from "../chat";
+import type { AilaDocumentContent } from "./types";
 
 const log = aiLogger("aila:lesson");
 
 export class AilaDocument implements AilaDocumentService {
   private readonly _aila: AilaServices;
-  private _plan: LooseLessonPlan;
-  private _hasSetInitialState = false;
+  private _content: AilaDocumentContent;
+  private _hasInitialisedContentFromMessages = false;
   private readonly _appliedPatches: ValidPatchDocument[] = [];
   private readonly _invalidPatches: ValidPatchDocument[] = [];
   private readonly _categoriser: AilaCategorisationFeature;
 
   constructor({
     aila,
-    lessonPlan,
+    content,
     categoriser,
   }: {
     aila: AilaServices;
-    lessonPlan?: LooseLessonPlan;
+    content?: LooseLessonPlan;
     categoriser?: AilaCategorisationFeature;
   }) {
-    log.info("Creating AilaLesson", lessonPlan?.title);
+    log.info("Creating AilaDocument");
     this._aila = aila;
-    this._plan = lessonPlan ?? {};
+    this._content = content ?? {};
     this._categoriser =
       categoriser ??
       new AilaCategorisation({
         aila,
       });
   }
 
-  public get plan(): LooseLessonPlan {
-    return this._plan;
+  public get content(): AilaDocumentContent {
+    return this._content;
   }
 
-  public set plan(plan: LooseLessonPlan) {
-    this._plan = plan;
+  public set content(content: AilaDocumentContent) {
+    this._content = content;
   }
 
-  public setPlan(plan: LooseLessonPlan) {
-    this._plan = plan;
+  public setContent(content: AilaDocumentContent) {
+    this._content = content;
   }
 
-  public get hasSetInitialState(): boolean {
-    return this._hasSetInitialState;
-  }
-
-  public set hasSetInitialState(value: boolean) {
-    this._hasSetInitialState = value;
+  public get hasInitialisedContentFromMessages(): boolean {
+    return this._hasInitialisedContentFromMessages;
```

Comment:
I've always wondered if these could be defined as setters and getters. But changing something like that would be out of scope for a rename like this so let's ignore this comment!

Reply:
I agree on this - we seem to have setPlan and also set plan. Probably quite easy to remove the setPlan method!